mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/AnnotationMapperTest.java

```diff
@@ -108,10 +109,31 @@ public void getAllAnnotationsShouldRetrieveStoredAnnotations() throws MailboxExc
     @ContractTest
     public void getAnnotationsByKeysShouldReturnStoredAnnotationWithFilter() throws MailboxException {
         annotationMapper.insertAnnotation(PRIVATE_ANNOTATION);
-        annotationMapper.insertAnnotation(PRIVATE_USER_ANNOTATION);
+        annotationMapper.insertAnnotation(PRIVATE_CHILD_ANNOTATION);
 
         assertThat(annotationMapper.getAnnotationsByKeys(ImmutableSet.of("/private/comment")))
             .containsOnly(PRIVATE_ANNOTATION);
     }
 
+    @ContractTest
+    public void getAnnotationsByKeysWithOneDepthShouldReturnThatEntryAndItsChildren() throws MailboxException {
+        annotationMapper.insertAnnotation(PRIVATE_ANNOTATION);
+        annotationMapper.insertAnnotation(PRIVATE_CHILD_ANNOTATION);
+        annotationMapper.insertAnnotation(PRIVATE_GRANDCHILD_ANNOTATION);
+        annotationMapper.insertAnnotation(SHARED_ANNOTATION);
+
+        assertThat(annotationMapper.getAnnotationsByKeysWithOneDepth(ImmutableSet.of("/private/comment")))
+            .contains(PRIVATE_ANNOTATION, PRIVATE_CHILD_ANNOTATION);
+    }
+
+    @ContractTest
+    public void getAnnotationsByKeysWithAllDepthShouldReturnThatEntryAndAllBelowEntries() throws MailboxException {
+        annotationMapper.insertAnnotation(PRIVATE_ANNOTATION);
+        annotationMapper.insertAnnotation(PRIVATE_CHILD_ANNOTATION);
+        annotationMapper.insertAnnotation(PRIVATE_GRANDCHILD_ANNOTATION);
+        annotationMapper.insertAnnotation(SHARED_ANNOTATION);
+
+        assertThat(annotationMapper.getAnnotationsByKeysWithAllDepth(ImmutableSet.of("/private/comment")))
+            .contains(PRIVATE_ANNOTATION, PRIVATE_CHILD_ANNOTATION, PRIVATE_GRANDCHILD_ANNOTATION);
+    }
```

Comment:
What happens if I have an annotation that have a key /private/commen ?
/private/commentextra ?

You do not test also edge case with empty set.
Idem for several entries in the set of requested keys.
What happen if I request a key that do not exist and do not have clilds ?
What happen if I ask about a key that do not exist but have childs (eg: /private ) ?